src/compiler/scanner.ts

```diff
@@ -293,17 +294,18 @@ const charToRegExpFlag = new Map(Object.entries({
     y: RegularExpressionFlags.Sticky,
 }));
 
-const regExpFlagToFirstAvailableLanguageVersion = new Map([
-    [RegularExpressionFlags.HasIndices, ScriptTarget.ES2022],
-    [RegularExpressionFlags.Global, ScriptTarget.ES3],
-    [RegularExpressionFlags.IgnoreCase, ScriptTarget.ES3],
-    [RegularExpressionFlags.Multiline, ScriptTarget.ES3],
-    [RegularExpressionFlags.DotAll, ScriptTarget.ES2018],
-    [RegularExpressionFlags.Unicode, ScriptTarget.ES2015],
-    [RegularExpressionFlags.UnicodeSets, ScriptTarget.ESNext],
-    [RegularExpressionFlags.Sticky, ScriptTarget.ES2015],
+const regExpFlagToFirstAvailableLanguageVersion = new Map<RegularExpressionFlags, LanguageFeatureMinimumTarget>([
+    [RegularExpressionFlags.HasIndices, LanguageFeatureMinimumTarget.RegularExpressionFlagsHasIndices],
+    [RegularExpressionFlags.DotAll, LanguageFeatureMinimumTarget.RegularExpressionFlagsDotAll],
+    [RegularExpressionFlags.Unicode, LanguageFeatureMinimumTarget.RegularExpressionFlagsUnicode],
+    [RegularExpressionFlags.UnicodeSets, LanguageFeatureMinimumTarget.RegularExpressionFlagsUnicodeSets],
+    [RegularExpressionFlags.Sticky, LanguageFeatureMinimumTarget.RegularExpressionFlagsSticky],
 ]);
 
+function getRegExpFlagToFirstAvailableLanguageVersion(flag: RegularExpressionFlags) {

```

Comment:
Instead of this may be a change this function to do:
```ts
const availableFrom = getRegExpFlagToFirstAvailableLanguageVersion(flag);
                        if (availableFrom && languageVersion < availableFrom) {
                            error(Diagnostics.This_regular_expression_flag_is_only_available_when_targeting_0_or_later, p, 1, getNameOfScriptTarget(availableFrom));
                        }
```